frontend/src/views/CellGuide/common/utils.ts

```diff
@@ -0,0 +1,30 @@
+import { TISSUE_DESCENDANTS } from "src/components/common/Filter/common/constants";
+
+export function isTissueIdDescendantOfAncestorTissueId(
+  tissueId: string,
+  ancestorTissueId: string
+): boolean {
+  // a tissue is a descendant of itself
+  if (tissueId === ancestorTissueId) return true;
+
+  // check all descendants of ancestorTissueId excluding ancestorTissueId itself
+  const descendantTermIds = TISSUE_DESCENDANTS[ancestorTissueId];
+  if (!descendantTermIds) return false;
+
+  return descendantTermIds.includes(tissueId);
+}
+
+export function filterDescendantsOfAncestorTissueId(
+  tissueIdList: string[],
+  ancestorTissueId: string
+): string[] {
+  const descendantList = [ancestorTissueId];
+
+  const moreDescendantTermIds = TISSUE_DESCENDANTS[ancestorTissueId];
+  if (moreDescendantTermIds) {
+    descendantList.push(...moreDescendantTermIds);
+  }
+  const discendantSet = new Set(descendantList);
+
+  return [...tissueIdList].filter((value) => discendantSet.has(value));
```

Comment:
You don't need the spread operation here, just do `tissueIdList.filter`

Reply:
🤦 will fix...I lifted this straight from perplexity.ai. I should remind myself that I should still think through pronouncements of the AI gods